src/main/java/org/opensearch/knn/index/util/NativeLibrary.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
```

Comment:
We need to have shorter 2-lines license header, please check other classes from PR as well

Reply:
We should update the template. Not in this PR.